File: BOJ/17609_String.cpp
#include <iostream>
#include <string>
#include <vector>

// boj 17609 회문, 문자열, 실버 1
using namespace std;

vector<int> answer;

int check(string str){
    int start = 0, end = str.size()-1;
    int result = 0;

    while (start<=end){
        if (str[start] == str[end]){
            start++;
            end--;
        }else{
            int left = start+1, right = end;
            bool left_pass = true;
            while (left<=right){
                if(str[left] !=  str[right]){
                    left_pass = false;
                    break;
                }
                left++;
                right--;
            }

            if (left_pass){
                result = 1;
                return result;
            }

            left = start, right = end-1;
            bool right_pass = true;
            while (left<=right){
                if (str[left] != str[right]){
                    right_pass = false;
                    break;
                }
                left++;
                right--;
            }

            if (right_pass) result = 1;
            else result = 2;

            return result;
        }
    }
    return result;
}
int main() {
    int T;
    cin>>T;

    answer.assign(T, -1);
    for (int i = 0; i < T; ++i) {
        string input;
        cin>>input;
        answer[i] = check(input);
    }

    for (int i = 0; i < T; ++i) {
        cout<<answer[i]<<"\n";
    }
    return 0;
}

File: BOJ/17609_String.swift
import Foundation
// boj 17609 회문, 문자열, 골드 5

let T = Int(readLine()!)!

var answer = ""

for _ in 0..<T {
    let word = readLine()!.map{String($0)}
    var result = 0
    var left = 0, right = word.count-1
    
    while left < right {
        if word[left] == word[right] {
            left += 1
            right -= 1
        } else {
            var nLeft = left + 1
            var nRight = right
            
            while nLeft < nRight {
                if word[nLeft] ==  word[nRight] {
                    nLeft += 1
                    nRight -= 1
                } else {
                    result += 1
                    break
                }
            }
            
            nLeft = left
            nRight = right-1
            
            while nLeft < nRight {
                if word[nLeft] ==  word[nRight] {
                    nLeft += 1
                    nRight -= 1
                } else {
                    result += 1
                    break
                }
            }
            
            if result == 0 { result += 1}
            break
        }
    }
    answer += "\(result)\n"
}

print(answer)

